docs/serving/autoscaling.md

```diff
@@ -179,11 +182,26 @@ template.
 spec:
   template:
     metadata:
-      autoscaling.knative.dev/metric: concurrency
+      autoscaling.knative.dev/metric: cpu
       autoscaling.knative.dev/class: hpa.autoscaling.knative.dev
 ```
 
-## Additional resources
+# Configuring the PodAutoscaler custom resource
+
+You can create a reconciler that operates on your own class of Knative PodAutoscaler by copying a [Kubernetes sample controller](https://github.com/kubernetes/sample-controller) and modifying its configuration to suit your desired use case.
+
+For example, if your service yaml includes a class annotation of:
+```
+autoscaling.knative.dev/class: sample
+```
+You would target that class annotation in your controller. For example:
+```
+    if pa.Annotations["autoscaling.knative.dev/class"] == "sample" {
+      // Do this
+    }
+```
+
```

Comment:
Let's add a few sentences here that one must implement the different mechanisms of the autoscaler to successfully make it work in Knative. Most importantly:

1. It'd make sense to use the autoscaling-base-reconciler as implemented in Knative Serving (link to https://github.com/knative/serving/blob/master/pkg/reconciler/autoscaling/reconciler.go).
2. Make sure to call `ReconcileSKS` from that reconciler.
3. If you want to use the metrics that Knative collects (concurrency metrics), make sure to use `ReconcileMetric` and `ReconcileMetricsService` to enable that system.

A complete example can be found in the HPA reconciler which is essentially an implementation of this: https://github.com/knative/serving/blob/master/pkg/reconciler/autoscaling/hpa/hpa.go

---

Let's link to these resources for now and just mention them. We have on our roadmap to provide an entire example implementation of this, with comments and all. We can revisit these docs once we've done that.

Reply:
Added new section to beginning of doc, lmk what you think

Reply:
I don't think it fares well above all the config. This really only applies to implementing an own Autoscaler.